dvc/scm/git/tree.py

```diff
@@ -13,6 +13,14 @@
 GIT_MODE_FILE = 0o644
 
 
+def _iter_tree(tree):
+    submodules = {x.hexsha: x for x in tree.repo.submodules}
```

Comment:
Let's add a note here describing what we are doing and why.

Reply:
Done